[receiver/sqlquery] TestLogsQueryReceiver_Collect is failing on Windows

Seen in:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5855756531/job/15874169475
- https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5856539541/job/15876482155

```
--- FAIL: TestLogsQueryReceiver_Collect (0.00s)
    logs_receiver_test.go:44: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sqlqueryreceiver/logs_receiver_test.go:44
        	Error:      	Should not be: 0x177b4147942c5704
        	Test:       	TestLogsQueryReceiver_Collect
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver	0.289s
FAIL
mingw32-make[2]: *** [../../Makefile.Common:100: test] Error 1
```

--- a/receiver/sqlqueryreceiver/logs_receiver_test.go
+++ b/receiver/sqlqueryreceiver/logs_receiver_test.go
@@ -41,7 +41,7 @@ func TestLogsQueryReceiver_Collect(t *testing.T) {
 
 	logRecord = logs.ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().At(1)
 	assert.Equal(t, "63", logRecord.Body().Str())
-	assert.NotEqual(t, logRecord.ObservedTimestamp(), pcommon.NewTimestampFromTime(now))
+	assert.GreaterOrEqual(t, logRecord.ObservedTimestamp(), pcommon.NewTimestampFromTime(now))
 
 	assert.Equal(t,
 		logs.ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().At(0).ObservedTimestamp(),

